packages/grid/x-data-grid-generator/package.json

```
@@ -41,7 +41,7 @@
   },
   "peerDependencies": {
     "@mui/icons-material": "^5.2.5",
-    "@mui/material": "^5.2.7",
+    "@mui/material": "^5.2.8",
     "react": "^17.0.2"
```

Same as above, this is a BC from #2419.

```suggestion
    "react": "^17.0.0"
```

---

x all the other peer dependency instances.

---

Related: https://github.com/mui-org/material-ui/pull/29283